bitcash/network/services.py

```diff
@@ -11,209 +12,56 @@
-        r = requests.post(cls.MAIN_TX_PUSH_API, json={
-                          cls.TX_PUSH_PARAM: tx_hex, 'network': 'mainnet', 'coin': 'BCH'}, timeout=DEFAULT_TIMEOUT)
-        return True if r.status_code == 200 else False
-
-
-class BitcoinDotComAPI():
+class BitcoinDotComAPI:
     """ rest.bitcoin.com API """
-    MAIN_ENDPOINT = 'https://rest.bitcoin.com/v2/'
-    MAIN_ADDRESS_API = MAIN_ENDPOINT + 'address/details/{}'
-    MAIN_UNSPENT_API = MAIN_ENDPOINT + 'address/utxo/{}'
-    MAIN_TX_PUSH_API = MAIN_ENDPOINT + 'rawtransactions/sendRawTransaction/{}'
-    MAIN_TX_API = MAIN_ENDPOINT + 'transaction/details/{}'
-    MAIN_TX_AMOUNT_API = MAIN_TX_API
-    MAIN_RAW_API = MAIN_ENDPOINT + 'transaction/details/{}'
-    TX_PUSH_PARAM = 'rawtx'
 
-    TEST_ENDPOINT = 'https://trest.bitcoin.com/v2/'
-    TEST_ADDRESS_API = TEST_ENDPOINT + 'address/details/{}'
-    TEST_UNSPENT_API = TEST_ENDPOINT + 'address/utxo/{}'
-    TEST_TX_PUSH_API = TEST_ENDPOINT + 'rawtransactions/sendRawTransaction/{}'
-    TEST_TX_API = TEST_ENDPOINT + 'transaction/details/{}'
-    TEST_TX_AMOUNT_API = TEST_TX_API
-    TEST_RAW_API = TEST_ENDPOINT + 'transaction/details/{}'
-
-    REGTEST_ENDPOINT = 'http://localhost:12500/v2/'
-    REGTEST_ADDRESS_API = REGTEST_ENDPOINT + 'address/details/{}'
-    REGTEST_UNSPENT_API = REGTEST_ENDPOINT + 'address/utxo/{}'
-    REGTEST_TX_PUSH_API = REGTEST_ENDPOINT + 'rawtransactions/sendRawTransaction/{}'
-    REGTEST_TX_API = REGTEST_ENDPOINT + 'transaction/details/{}'
-    REGTEST_TX_AMOUNT_API = REGTEST_TX_API
-    REGTEST_RAW_API = REGTEST_ENDPOINT + 'transaction/details/{}'
+    NETWORK_ENDPOINTS = {
+        'mainnet': os.getenv("BITCOINCOM_API_MAINNET") or 'https://rest.bitcoin.com/v2/',
+        'testnet': os.getenv("BITCOINCOM_API_TESTNET") or 'https://trest.bitcoin.com/v2/',
+        'regtest': os.getenv("BITCOINCOM_API_REGTEST") or 'http://localhost:12500/v2/'
+    }
+    UNSPENT_PATH = 'address/utxo/{}'
+    ADDRESS_PATH = 'address/details/{}'
+    RAW_TX_PATH = 'rawtransactions/sendRawTransaction/{}'
+    TX_DETAILS_PATH = 'transaction/details/{}'
 
-    @classmethod
-    def get_balance(cls, address):
-        r = requests.get(cls.MAIN_ADDRESS_API.format(address),
-                         timeout=DEFAULT_TIMEOUT)
-        r.raise_for_status()  # pragma: no cover
-        data = r.json()
-        balance = data['balanceSat'] + data['unconfirmedBalanceSat']
-        return balance
+    TX_PUSH_PARAM = 'rawtx'
 
     @classmethod
-    def get_balance_testnet(cls, address):
-        r = requests.get(cls.TEST_ADDRESS_API.format(address),
-                         timeout=DEFAULT_TIMEOUT)
-        r.raise_for_status()  # pragma: no cover
-        data = r.json()
-        balance = data['balanceSat'] + data['unconfirmedBalanceSat']
-        return balance
+    def network_endpoint(cls, network):
+        if network not in NETWORKS:
+            raise InvalidNetwork("No endpoints found for network {}".format(network))
```

Comment:
You can use f-strings, like `f"No endpoints found for network {network}"`

Reply:
I think f-strings were added in [Python 3.6](https://docs.python.org/3/whatsnew/3.6.html). We can use them, but will have to drop 3.5 support. Should we drop it or skip f-strings for now @teran-mckinney?

Reply:
Unless there's a lot of objections, I'd like to jump over to 3.7. 3.9 is the latest stable version as it is. 3.5 is pretty old, Debian 9 era.

Reply:
Well f-strings seem pretty cool. Since this is a refactor PR, how about I go ahead and put them [everywhere](https://github.com/pybitcash/bitcash/pull/86/commits/5e057a0ed453e349b8eedc28f8fe89c07d57c1b5).

Reply:
Very nice!!